pkg/controller/external_secrets_manager/controller.go

```diff
@@ -0,0 +1,247 @@
+	}
+
+	return r.processReconcileRequest(esm)
+}
+
+// processReconcileRequest is the reconciliation handler to manage the resources.
+func (r *Reconciler) processReconcileRequest(esm *operatorv1alpha1.ExternalSecretsManager) (ctrl.Result, error) {
+	statusUpdated := false
+	if esm.Status.ControllerStatuses == nil {
+		esm.Status.ControllerStatuses = make([]operatorv1alpha1.ControllerStatus, 0)
+	}
+	if r.externalSecrets != nil && len(r.externalSecrets.Status.Conditions) > 0 {
+		for _, esCond := range r.externalSecrets.Status.Conditions {
+			if r.updateStatusCondition(esm, externalSecretsControllerId, esCond) {
+				statusUpdated = true
+			}
+		}
+	}
+
+	if statusUpdated {
+		if err := r.updateStatus(r.ctx, esm); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to update externalsecretsmanager.openshift.operator.io status: %w", err)
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// getControllerCondition returns the status condition of a specific controller.
+func getControllerCondition(controllerName string, esm *operatorv1alpha1.ExternalSecretsManager) *operatorv1alpha1.ControllerStatus {
+	for i, s := range esm.Status.ControllerStatuses {
+		if s.Name == controllerName {
+			return &esm.Status.ControllerStatuses[i]
+		}
+	}
+	status := &operatorv1alpha1.ControllerStatus{
+		Name:       controllerName,
+		Conditions: make([]operatorv1alpha1.Condition, 0),
+	}
+	esm.Status.ControllerStatuses = append(esm.Status.ControllerStatuses, *status)
+	return status
+}
+
+// updateStatusCondition updates the status condition of a specific controller.
+func (r *Reconciler) updateStatusCondition(esm *operatorv1alpha1.ExternalSecretsManager, controllerName string, updCondition metav1.Condition) bool {
+	status := getControllerCondition(controllerName, esm)
+
+	found, condUpdated := false, false
+	for i, c := range status.Conditions {
+		if c.Type == updCondition.Type {
+			found = true
+			if c.Status != updCondition.Status || c.Message != updCondition.Message {
+				status.Conditions[i].Status = updCondition.Status
+				status.Conditions[i].Message = updCondition.Message
+				condUpdated = true
+			}
+		}
+	}
+
+	if !found {
+		status.Conditions = append(status.Conditions, operatorv1alpha1.Condition{
+			Type:    updCondition.Type,
+			Status:  updCondition.Status,
+			Message: updCondition.Message,
+		})
+		condUpdated = true
+	}
+
+	status.ObservedGeneration = updCondition.ObservedGeneration
+	esm.Status.LastTransitionTime = metav1.Now()
+
+	return condUpdated
+}
+
+// updateStatus is for updating the status subresource of externalsecretsmanager.openshift.operator.io.
+func (r *Reconciler) updateStatus(ctx context.Context, changed *operatorv1alpha1.ExternalSecretsManager) error {
+	namespacedName := types.NamespacedName{Name: changed.Name, Namespace: changed.Namespace}
+	if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
we should plan to refactor the updateStatus with a wrapped retryOnConflict,
that retries with a known good backoff limits. However, it is not a must for this PR and just something we can generally put on the client itself across ES and ESM.